Dockerfile

```diff
@@ -0,0 +1,45 @@
+# This file is for use as a devcontainer and a runtime container
+# 
+# The devcontainer should use the build target and run as root with podman 
+# or docker with user namespaces.
+#
+FROM python:3.10 as build
+
+# Add any system dependencies for the developer/build environment here
+RUN apt-get update && apt-get upgrade -y && \
+    apt-get install -y --no-install-recommends \
+    build-essential \
+    busybox \
+    git \
+    net-tools \
+    vim \
+    && rm -rf /var/lib/apt/lists/* \
+    && busybox --install
+
+COPY . /project
+
+RUN cd /project && \
+    pip install --upgrade pip build && \
+    export SOURCE_DATE_EPOCH=$(git log -1 --pretty=%ct) && \
+    python -m build --sdist --wheel && \
+    touch requirements.txt
+
+RUN python -m venv /venv
+ENV PATH=/venv/bin:$PATH
+
+RUN cd /project && \
+    pip install --upgrade pip && \
+    pip install -r requirements.txt dist/*.whl && \
+    pip freeze  > requirements.txt && \
+    sed -i '/file:\/\//d' requirements.txt
+
+FROM python:3.10-slim as runtime
+
+# Add apt-get system dependecies for runtime here if needed
+
+COPY --from=build /venv/ /venv/
+ENV PATH=/venv/bin:$PATH
```

Comment:
It does this twice?

Reply:
YES once in each target. ENV is scoped at the current target level. I could set a global ARG and then set each ENV to its value to avoid repetition but I preferred the simpler option of repeating.